pallets/subtensor/src/subnets/weights.rs

```diff
@@ -105,6 +106,77 @@ impl<T: Config> Pallet<T> {
         })
     }
 
+    /// ---- The implementation for the extrinsic batch_commit_weights.
+    ///
+    /// This call runs a batch of commit weights calls, continuing on errors.
+    ///
+    /// # Args:
+    ///  * 'origin': (<T as frame_system::Config>RuntimeOrigin):
+    ///    - The signature of the calling hotkey.
+    ///
+    ///  * 'netuids' ( Vec<Compact<u16>> ):
+    ///    - The u16 network identifiers.
+    ///
+    ///  * 'commit_hashes' ( Vec<H256> ):
+    ///    - The commit hashes to be committed, one hash for each netuid in the batch.
+    ///
+    /// # Event:
+    ///  * WeightsCommitted;
+    ///    - On successfully storing the weight hashes.
+    ///  * BatchCompletedWithErrors;
+    ///    - Emitted when at least on of the weight commits has an error.
+    ///  * BatchWeightItemFailed;
+    ///    - Emitted for each error within the batch.
+    ///  * BatchWeightsCompleted
+    ///    - Emitted when the batch of weights is completed.
+    ///
+    pub fn do_batch_commit_weights(
+        origin: T::RuntimeOrigin,
+        netuids: Vec<Compact<u16>>,
+        commit_hashes: Vec<H256>,
+    ) -> dispatch::DispatchResult {
+        // --- 1. Check the caller's signature. This is the hotkey of a registered account.
+        let hotkey = ensure_signed(origin.clone())?;
+        log::debug!(
+            "do_batch_commit_weights( origin:{:?} netuids:{:?}, hashes:{:?}",
+            hotkey,
+            netuids,
+            commit_hashes
+        );
+
+        let results: Vec<dispatch::DispatchResult> = netuids
+            .iter()
+            .zip(commit_hashes.iter())
```

Comment:
If commit_hashes vector is longer than netuids vector, we will be committing weights for root, which will fail. Is this intentional that we don't check for vector lengths to match?

Reply:
added a check